pkg/apis/workflow/v1alpha1/container_set_template_types.go

```diff
@@ -12,16 +12,18 @@ import (
 type ContainerSetTemplate struct {
 	Containers   []ContainerNode      `json:"containers" protobuf:"bytes,4,rep,name=containers"`
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty" protobuf:"bytes,3,rep,name=volumeMounts"`
-	// RetryStrategy describes how to retry a container nodes in the container set if it fails.
-	// Nbr of retries(default 0) and sleep duration between retries(default 0s, instant retry) can be set.
+	// RetryStrategy describes how to retry container nodes if the container set fails.
+	// Note that the container set will not be retried if a container's `command` cannot be located, as it will fail each time.
```

Comment:
```suggestion
	// Note that this works differently from the template-level `retryStrategy` as it is a process-level retry that does not create new Pods or containers.
```

changing this to account for the other differences. we can include more implementation details like "process-level retry" in API docs/comments. And don't need to be very specific here, as the code and the "Note" in the docs go into the rest